anndata/_io/h5ad.py

```diff
@@ -128,10 +135,13 @@ def write_sparse_as_dense(f, key, value, dataset_kwargs=MappingProxyType({})):
         del f[key]
 
 
-def read_h5ad_backed(filename: Union[str, Path], mode: Literal["r", "r+"]) -> AnnData:
-    d = dict(filename=filename, filemode=mode)
+def read_h5ad_backed(
+    file: Union[str, Path, BinaryIO], mode: Literal["r", "r+"]
+) -> AnnData:
+    file_name = Path(file) if isinstance(file, (str, Path)) else getattr(file, "name")
+    d = dict(filename=file_name, filemode=mode)
```

Comment:
attempt to get the name here. until now we’d just add the file object as `filename`